### PR TITLE
2pc: MM functionality

### PR DESCRIPTION
**Pull request by sougou**
* New proto var TransactionMetadata.
* Use a constant for long (forever) timeouts.
* Core MM functionality. A few more will be added later.

### CODE REVIEW

File: go/vt/tabletserver/config.go [hunk 1]
```diff
@@ -20,6 +21,9 @@ import (
 	topodatapb "github.com/youtube/vitess/go/vt/proto/topodata"
 )
 
+// longTime is for setting long timeouts, practically forever.
+const longTime = 24 * 365 * 10 * time.Hour
```
Comment (alainjobart) [inline go/vt/tabletserver/config.go]: I think I'd rather use 0 in this case, and in execRequest have:
if timeout != 0 {
  var cancel context.CancelFunc
  ctx, cancel = context.WithTimeout(timeout)
  defer cancel()
}

What do you think?

File: proto/query.proto [hunk 2]
```diff
@@ -523,3 +523,20 @@ message UpdateStreamRequest {
...
+
+// TransactionState represents the state of a distributed transaction.
+enum TransactionState {
+  UNKNOWN = 0;
+  PREPARE = 1;
+  COMMIT = 2;
+  ROLLBACK = 3;
+}
+
+// TransactionMetadata contsins the metadata for a distributed transaction.
```
Comment (alainjobart) [inline proto/query.proto]: /contsins/contains/

File: go/vt/tabletserver/config.go (hunk 1 above)
Comment (dumbunny) [inline go/vt/tabletserver/config.go]: I agree with Alain here. From the value, there's a subtle implication here that Vitess supports very long streaming queries, when there are numerous reasons to discourage day-long queries. I'd prefer using a reserved value that doesn't have an obvious time-scale meaning.

**Comment by sougou** [comment]
LGTM
Code review comments addressed.

**Pull request merged**